src/vcpkg/commands.build.cpp

```diff
@@ -1488,55 +1488,84 @@ namespace vcpkg
         return res;
     }
 
-    struct CreateLogDetails
+    void append_log(const Path& path, const std::string& log, int max_size, std::string& out)
     {
-        const Filesystem& fs;
+        StringLiteral details_start = "<details><summary>{}</summary>\n\n```\n";
+        StringLiteral skipped_msg = "\n...\nSkipped {} lines\n...";
+        StringLiteral details_end = "\n```\n</details>";
+        max_size -= static_cast<int>(path.native().size() + details_start.size() + details_end.size() +
+                                     skipped_msg.size() + 6 /* digits for skipped count */);
+        if (max_size < 100)
+        {
+            return;
+        }
```

Comment:
This is still vulnerable due to doing mixed `int`/`size_t` operations.

```suggestion
        const size_t context_size = path.native().size() + details_start.size() + details_end.size() + skipped_msg.size() + 6 /* digits for skipped count */;
        static constexpr size_t minimum_log_size = 100;
        const size_t minimum_usable_size = context_size + minimum_log_size;
        if (static_cast<size_t>(max_size) < minimum_usable_size)
        {
            return;
        }
        
        // cast is safe because `max_size`, an int, is bigger
        max_size -= static_cast<int>(context_size);
```

Can we just make these all `size_t`s so we can stop worrying about this? There's no reason for `max_size` to ever be negative.